spec.bs

```diff
@@ -1989,6 +1997,80 @@ finalized.
+
+Note: A [=browsing context=]'s [=browsing context/fenced frame config instance=] is assigned in two
+different ways: (1) during embedder-initiated <{fencedframe}> navigation, as described by the rest
+of this section, and (2) inherited during initial {{Document}} creation, for same-origin
+{{Document}}s in [=child navigables=] whose [=navigable/traversable navigable=] is a [=fenced
+navigable container/fenced navigable=], as handled in the [[#creating-browsing-contexts-patch]]
+section.
+
+For any successful navigation inside of a <{fencedframe}>, regardless of whether it was initiated by
+content in the <{fencedframe}> or its embedder, exactly one of three things will happen:
+
+ * <i>Embedder-initiated navigations</i>: The [=navigation params=]'s [=navigation params/fenced
+   frame config instance=] will be unconditionally assigned to the <{fencedframe}>'s [=fenced
+   navigable container=]'s [=fenced navigable container/fenced navigable=]'s [=navigable/active
+   browsing context=]'s [=browsing context/fenced frame config instance=].
+
+ * <i>Cross-origin inner-content-initiated navigations</i>: We do not want the <{fencedframe}>'s
+   [=fenced navigable container=]'s [=fenced navigable container/fenced navigable=]'s
+   [=navigable/active browsing context=]'s [=browsing context/fenced frame config instance=] to
+   persist through this kind of cross-{{Document}} same-[=browsing context group=] navigation, so
+   we manually clear it.
+
+ * <i>Same-origin inner-content-initiated navigations</i>: The [=fenced frame config instance=]
+   referenced in the immediately-preceding point persists through this kind of cross-{{Document}}
+   navigation. The patches below take no action to clear it, so it continues to be visible through
+   the {{Window/fence}} API.
+
+<div algorithm=create-and-initialize-document-patch>
+  Modify [[!HTML]]'s [=create and initialize a Document object=] algorithm to insert a step after
+  step 2 that reads:
+
+  3. If |navigationParams|'s [=navigation params/fenced frame config instance=] is not null:
+
+    1. [=Assert=]: |browsingContext| does not equal |navigationParams|'s [=navigation
+       params/navigable=]'s [=navigable/active browsing context=].
+
+       Note: This is because embedder-initiated navigations (indicated by |navigationParams|'s
+       [=navigation params/fenced frame config instance=] being non-null) always cause a
+       [[#bcg-swap]].
+
+    1. Set |browsingContext|'s [=browsing context/fenced frame config instance=] to
+       |navigationParams|'s [=navigation params/fenced frame config instance=].
+
+  4. Otherwise, if |navigationParams|'s [=navigable/active browsing context=]'s [=browsing
+     context/fenced frame config instance=] is non-null, and either of the following conditions are
+     true:
+
+       * |navigationParams|'s [=navigation params/origin=] is not [=same origin=] with
+         |navigationParams|'s [=navigation params/navigable=]'s [=navigable/active document=]'s
```

Comment:
This should be whether it's same-origin with the config instance's mapped url, not the active document.

Reply:
Hmm it sounds like the second half of this whole section should probably just be deleted given https://github.com/WICG/fenced-frame/pull/101#discussion_r1218282328. Do you agree?